src/Caret.tsx

```diff
@@ -56,12 +56,15 @@ export type CaretProps = {
   borderWidth?: string | number
   size?: number
   location?: Location
+  theme?: any
 }
 
 function Caret(props: CaretProps) {
-  const {bg} = getBg(props)
-  const {borderColor} = getBorderColor(props)
-  const {borderWidth} = getBorderWidth(props)
+  const theme = React.useContext(ThemeContext)
+  const propsWithTheme = {...props, theme: props.theme ?? theme}
```

Comment:
Can you explain why this is necessary? If the theme doesn't exist in `props` why would `ThemeContext` still work?

Reply:
Since Caret is not a styled component, we need to explicitly get the theme from context and use it as a fallback when evaluating the system props in the lines below (i.e. `getBg()`).